### ISSUE TITLE
Fails if `(homepage|license).workspace = true` is used

### ISSUE THREAD

**Issue by ionenwks** [projg2/pycargoebuild#17]
e.g. with app-emulation/ruffle

desktop/Cargo.toml has:
```
[package]
name = "ruffle_desktop"
authors.workspace = true
edition.workspace = true
homepage.workspace = true
license.workspace = true
repository.workspace = true
version.workspace = true
```

then
```
$ pycargoebuild desktop
DEBUG(cargo_to_spdx): license_str = {'workspace': True}
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.11/pycargoebuild", line 8, in <module>
    sys.exit(entry_point())
             ^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/__main__.py", line 213, in entry_point
    sys.exit(main(*sys.argv))
             ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/__main__.py", line 121, in main
    pkg_metas.append(get_package_metadata(f))
                     ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/cargo.py", line 94, in get_package_metadata
    pkg_license = cargo_to_spdx(pkg_license)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pycargoebuild/cargo.py", line 55, in cargo_to_spdx
    return license_str.replace("/", " OR ")
           ^^^^^^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute 'replace'
```
(wrt DEBUG line, I added a quick print to see what it was set to)

Continues beyond if Cargo.toml line is removed but imagine it won't read top level Cargo.toml's license.

(still fails either way with `RuntimeError: Unsupported crate source: git+https://github.com/RustAudio/dasp?rev=f05a703#<snip>` wrt #15 -- wish it'd just ignore them like cargo-ebuild for now given I handle these semi-manually, would want to at least leverage the rest to pycargoebuild **Edit:** well, for now I'll try to have my bump script mangle the git crates out of Cargo.lock so it doesn't complaint)

**Comment by ionenwks** [comment]
>Edit: well, for now I'll try to have my bump script mangle the git crates out of Cargo.lock so it doesn't complaint

Well, turns out it similarly fails on homespace.workspace after that, did give me an empty LICENSE and HOMEPAGE but the non-git crates and their license were handled properly.

**Comment by mgorny** [comment]
I think this should work… except that now ruffle fails because one of the packages doesn't specify a license… sigh. How low quality can Cargo packages be?

**Comment by ionenwks** [comment]
Yep thanks, gives me an ebuild and LICENSE and HOMEPAGE are set.

...ofc after skipping nihav-* that is, which only has a top level COPYING and is not mentioned in Cargo.toml -- also need `LicenseRef-UFL-1.0 = UbuntuFontLicense-1.0` for the epaint crate which is not a SPDX identifier so I've only kept that addition locally.

Guess when auto-detection fails (license files, or specified only elsewhere), having a way to map could be handy (like a local configuration file that could have `nihav-core = MIT`)

**Comment by mgorny** [comment]
Indeed, that's a better idea than just printing warnings. Perhaps both: print warning about unknown license (including `license-file` stuff), plus support overrides via config.

**Comment by mgorny** [comment]
@ionenwks, I've just pushed some config support to git (see the README). Could you play with it a bit and see if that's helpful? I've been able to get ruffle without warnings with the following config:

```toml
[license-overrides]
nihav_codec_support = "MIT"
nihav_duck = "MIT"
nihav_core = "MIT"

[license-mapping]
"LicenseRef-UFL-1.0" = "UbuntuFontLicense-1.0"
```

(I don't think we should keep this mapping in ::gentoo since it's not "official")

**Comment by ionenwks** [comment]
Tried a bit and that makes it easy to use now, thanks. Added an extra entry too given one crate has existing but incomplete licenses for the junk it bundles.

And yeah, wrt UFL-1.0 I didn't want to add it to ::gentoo either for same reason, so this makes it easier to add locally now.

Do see last thing though, ruffle ebuild is generated with 4 workspaces: `pycargoebuild desktop exporter scanner tests`, which works fine but it resulted in:
```
LICENSE="
    || ( MIT Apache-2.0 )
    || ( MIT Apache-2.0 )
    || ( MIT Apache-2.0 )
    || ( MIT Apache-2.0 )
"
```
Aside from that, it looks corrects and lets me bump it.

**Comment by mgorny** [comment]
Good catch, thanks! This is fixed now and I'll release 0.10 shortly.
